pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -101,6 +103,16 @@ func secondaryResourceTypes() []SecondaryResourceType {
 	}
 }
 
+var supportedTypes = map[string]*resources.ReconcileType{
```

Comment:
Why is this PR touching Routes? How is it related to automatic ServiceMonitor creation?

Reply:
Oh, @Crumby I changed a bit logic for optimized performance and reduce API server access, also we need an additional option to enable/disable Watch objects and don't do this for all dynamic discovery API's